lib/v2/medieval-china/post.js

```diff
@@ -0,0 +1,52 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const baseUrl = 'https://medieval-china.club';
+    const { data: response } = await got(baseUrl);
+    const $ = cheerio.load(response);
+    const posts = $('section.list-layout div.post-item', '.container')
+        .toArray()
+        .map((item) => {
+            item = $(item);
+            const pageUrl = item.find('div.post-wrap a').first().attr('href');
+            const title = item.find('div.post-content-wrap h3.post-title a').first().text();
+            const pubDate = timezone(parseDate(item.find('div.post-info span.post-info-item').first().text()), +8);
+            const category = item
+                .find('ul.card-tag-list li a')
+                .toArray()
+                .map((item) => $(item).text());
+            return {
+                title,
+                link: `${baseUrl}${pageUrl}`,
+                pubDate,
+                category,
+            };
+        });
```

Comment:
Please use the embedded JSON `window.localPosts` instead.
https://docs.rsshub.app/joinus/new-rss/before-start#acquire-data

Remember to [slice it](https://github.com/DIYgod/RSSHub/blob/d7ec908673fe652debc23ed1ca7dd9558251b0e2/lib/v2/cctv/jx.js#L16).

Reply:
```js
    const posts = JSON.parse(
        $('script:contains("window.localPosts")')
            .text()
            .match(/window\.localPosts = JSON\.parse\('(.*)'\);/)[1]
    ).map((item) => ({
        title: item.title,
        ...and so on
    }));
```